### DISCUSSION TITLE
Login failure

### DISCUSSION THREAD

**Discussion by ismael1361** [appy-one/acebase-server#72] [General]
We are using the AceBase database and have noticed that some users are unable to log in. We have identified that the API authentication is using the query.filter('email', '==', credentials.email), which limits the list of users and filters them, leaving out the rest of the list and preventing login. Our database currently has 233 registered users, but only 203 of them are able to log in.

**Comment by donl** [comment]
`acebase-client` provides both a `db.auth.signIn()` and `db.auth.signInWithEmail()` for using the `username` or `email` respectfully.

Hopefully that helps.

**Comment by appy-one** [comment]
@ismael1361 Is this still an issue or has it been resolved in the meantime? Did @donl's comment help to sign in users with username only instead of with email address?

**Reply by ismael1361** [reply]
I'm still having problems. We identified that the issue is with the query method, which returns a limited number of child nodes from the referenced path. In other words, in the module itself that filters users by email, the query method only sees the first 200 users, ignoring the rest that are registered, and returning a failure as if the user doesn't exist. I had to replicate the entire module and modify the code to use ref.get({ include: ["*/email"] }) and filter the list instead of using the query method.

**Reply by appy-one** [reply]
That is impossible. There is an index on the email property that is used by the query filter. The index will find the user with given email address instantly, it does not loop through all users!

**Reply by ismael1361** [reply]
It may seem impossible, even I thought it was impossible, but it happened to me and I had many user complaints that they couldn't log into their accounts, as if the account didn't exist, and they couldn't even register again because an account with the same email already existed. After investigating more thoroughly, which took about three days, I identified the error in the query, replaced it with get, and it worked.

**Comment by appy-one** [comment]
I am not able to reproduce this. I have just tested with a script creating 5000 accounts, then signing into them without any issues.
Can you tell me a little bit about your setup? What version of `acebase-server` are you using, how are you deploying it (cloud service / server), are you running 1 server instance or multiple?

**Reply by ismael1361** [reply]
We are using the latest version (1.16.2) of acebase-server on a VPS with 8 CPU cores, 16GB of RAM, 12TB of traffic and 1TB of storage. The server is running on 8 instances using the IPC method.

**Reply by appy-one** [reply]
How is your cluster managed? using pm2, standard NodeJS clustering or with an AceBase IPC server?

**Reply by ismael1361** [reply]
Standard NodeJS clustering.

```javascript
require("dotenv").config();
const cluster = require("cluster");
const { cpus } = require("os");
const { AceBase } = require("acebase");
const { AceBaseServer } = require("acebase-server");

if(cluster.isMaster){
    console.log("Master %d is running", worker.process.pid);

    for (let i = 0; i < cpus().length; i++) {
        cluster.fork();
    }

    cluster.on("exit", (worker, code, signal) => {
        console.log("worker %d died (%s). restarting...", worker.process.pid, signal || code);
        cluster.fork();
    });

    const db = new AceBase(process.env.DBNAME);
}else{
    let conf = {
        logLevel: 'none', 
        host: '0.0.0.0', 
        port: parseInt(process.env.SERVER_PORT),
        maxPayloadSize: '50mb',
        authentication: {
            enabled: true,
            allowUserSignup: true,
            defaultAccessRule: 'auth',
            defaultAdminPassword: password
        }, 
        email: { send: sendRequestedEmail },
        transactions: { 
            log: false,
            maxAge: 3,
            noWait: false
        }
    };

    const server = new AceBaseServer(process.env.DBNAME, conf);
}
```

**Reply by appy-one** [reply]
I think this is going to be fixed with https://github.com/appy-one/acebase/pull/211
